napari/_qt/tests/test_advanced.py

```diff
@@ -0,0 +1,139 @@
+import numpy as np
+
+from napari.components import ViewerModel
+from napari._qt.qt_viewer import QtViewer
+
+
+def test_4D_5D_images(qtbot):
+    """Test adding 4D followed by 5D image layers to the viewer.
+
+    Intially only 2 sliders should be present, then a third slider should be
+    created.
+    ."""
+    np.random.seed(0)
+    viewer = ViewerModel()
+    view = QtViewer(viewer)
+    qtbot.addWidget(view)
```

Comment:
If I understand correctly, you're using two steps here because you need the `view` for the qtbot. However, I wonder if it's better to do

```python
viewer = napari.Viewer()
qtbot.addWidget(viewer.window.qt_viewer)
```

because that is the type of instantiation that we want to test... I'm :+0.5: on that so happy to hear some pushback, just a suggestion.

Reply:
yup - agreed, i'll switch this over for the `advanced` tests. It also makes it more clear these are testing the whole app as opposed to just the qt viewer and I might move and rename accordingly